lib/TopSitesFeed.jsm

```diff
@@ -340,6 +342,20 @@ this.TopSitesFeed = class TopSitesFeed {
     if (!this._tippyTopProvider.initialized) {
       await this._tippyTopProvider.init();
     }
+    if (this.store.getState().Prefs.values[SEARCH_SHORTCUTS_EXPERIMENT] && !this.searchShortcutsInitialized) {
+      // Populate the state with available search shortcuts
+      const searchShortcuts = [];
+      SEARCH_SHORTCUTS.forEach(shortcut => {
```

Comment:
Can you look through the available search engines in `Services.search.getEngines()` and match on `engine.identifier.match(shortcut.searchIdentifier)`

Reply:
Can you also add an observer for engine changes as well to cause a refresh?

Reply:
`engine.identifier` can be null (for added engines), so check before `.match`-ing otherwise we run into bugs like https://bugzilla.mozilla.org/show_bug.cgi?id=1482464